Add My Projects page

0xzachvai opened the pull request:
## What does this PR do and why?

Adds a new top level page for viewing a person's joined projects, as mentioned [on notion](https://juicebox.notion.site/Add-My-Projects-page-that-lists-projects-you-ve-bought-into-8083f44673304c07aa341f69bece117e) and issue #256 .

I setup the navbar link to only display if the user has a wallet connected, and to send them to the Projects page if they disconnect while on My Projects.

On Notion, there's also requests for including additional information for each project. I wasn't sure what designs for that might look like, so I left them off this PR for now.

## Screenshots

![20211231_152511](https://user-images.githubusercontent.com/94148670/147838226-c10c3808-4d66-4e51-9f6c-2564acdbf921.png)

## Acceptance checklist

- [x] I have evaluted the [Approval Guidelines](../../CONTRIBUTING.md#approval-guidelines) for this PR.

peripheralist commented:
hey @0xzachvai awesome PR! and thanks for the improvements to the Participant parsing and the `in` graph query operator. couple thoughts here:

1) we should be able to simplify the query by using the `Participants` property of the `Project` graph entity, so we don't have to make two sequential queries. that said, we've recently discovered some issues with how participants are being indexed. i'm hoping to have this solved once the new subgraph version is launched for the v1.1 contracts (in the next 1-2 weeks), but to avoid any added confusion in the app let's hold off on shipping any new features that rely on that until then

2) from a layout perspective, i'd prefer to see "My projects" as a tab on the existing projects page, rather than a new page. it'll keep the navbar smaller, and i suspect most people won't need sub categories for active/archived. on a related note, i'd be in favor of including an "archived" tag on projects in the list view

0xzachvai commented:
Hey @peripheralist , thank for the feedback!

1. I was trying do this in a single query, but I couldn't personally figure out a way to do so with the current graph setup. `Project_filter` doesn't support filtering by participants, only creator, and doing this (is this what you meant?):
```graphql
query {
  projects {
    participants (where: {
      wallet: "0x01f0530e47f85c1d1f71823c47e0f0ca0415c1f2"
    }){
      ...
    }
    ...
  }
}
```

still returns all projects, regardless of whether it contains a matching participant entity. Let me know if I'm just missing something, I'm still relatively new to GraphQL.

2. Completely agree, I'll make these changes.

peripheralist commented:
Looks like you're right @0xzachvai, I didn't realize the graph still doesn't support nested queries. Your solution works great for now. Let's still hold off on shipping this until the `Participants` indexing is solved—this is a priority to have fixed before v1.1 is shipped, so hopefully we can get this out with the new app version in 1-2 weeks!

aeolianeth commented on src/components/MyProjects/index.tsx:
```diff
@@ -0,0 +1,140 @@
+        <Trans>My Projects on Juicebox</Trans>
+      </h1>
+      <p style={{ marginBottom: 40, maxWidth: 800 }}>
+        <Trans>
```
**praise**: thanks for marking this string for translation! 👏

aeolianeth commented on src/components/MyProjects/index.tsx:
```diff
@@ -0,0 +1,140 @@
+      <p style={{ marginBottom: 40, maxWidth: 800 }}>
+        <Trans>
+          <InfoCircleOutlined />
+          These are all projects built on the Juicebox protocol that you're
```
**suggestion**:

```suggestion
          Juicebox projects that you've contributed to.
```

0xzachvai commented:
Okay, My Projects is now a tab under the Projects page. I set it up to only show when a wallet's connected, but I'm not sure that's better than just showing the tab with an empty grid (or hint text), so open to suggestions. In the process, I extracted the tab function into two components, `TabGroup` and `Tab` - figured they could be reused elsewhere, especially if the Project page is redone like recently mentioned on the Notion.

Also added that Archived tag to project cards, with a nifty little tooltip - can take a look in the preview build under Archived tab.

Lmk if you catch anything else I can include or improve - otherwise, I'll just keep this branch up to date with main until the indexing stuff is ready.

peripheralist commented:
Thanks for the patience here @0xzachvai 😄 The new subgraph and app version were shipped yesterday, so we're clear to dive back into this! Looks like we made some merge conflicts for you. Could you try getting this rebased against the latest `main` so we could play with it and get some design feedback? Let me know if the rebase gets too dense, I'm happy to help out

peripheralist commented:
hey @0xzachvai, are you interested in picking this up again? if not no worries, i'll take this and run with it by the end of the week if i haven't heard back. thanks for the great work so far!

0xzachvai commented:
Hey! It's been a really busy week offline, but I should be able to pick this back up today/tomorrow. Might need a bit of help rebasing the projects/index.ts, I'll ping you if anything

peripheralist commented:
all good! let me know if i can help

peripheralist commented:
only closing this because it's been rebuilt in a more current PR. appreciate your help with this @0xzachvai !
